deps/policy-engine/src/policy/rego.rs

```diff
@@ -2,28 +2,48 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-use async_trait::async_trait;
+use std::collections::HashMap;
+
+use regorus::Extension;
 use serde_json::Value;
-use tracing::instrument;
+use tracing::{info, instrument};
 
-use crate::{Engine, EvaluationResult, PolicyError};
+use crate::{EngineTrait, EvaluationResult, PolicyEngine, PolicyEngineConfig, PolicyError, Result};
 
 /// The rule to evaluate the policy.
 /// Note that only the result of this rule will be returned.
 pub const EVAL_RULE: &str = "data.policy.result";
 
+pub struct ExtensionWrapper {
```

Comment:
nit: I think you can just call this `RegorusExtension` (to distinguish it from EAR extensions)